Enabled CORS to allow calls from frontend to the backend api

SAbbas2018 opened the pull request:
## [DEV-031] Issue Title
<!--- Please add in the issue number this PR is fixing below. If an issue does not exist, create one! --->
Fixes: #31 

-   [Y] Has the code been formatted?
-   [Y] Have dependencies/packages been added?

### Changes
Enabled CORS in the backend router to allow for API calls made by the frontend from http://localhost:3000. 
Fixed the linting manually, the checkers now pass.
### Testing
-   [Y] Does the code have sufficient automated test coverage? If not, include a reasoning and steps required to manually QA
-   I checked by making multiple requests from the frontend and they all came back with response status 200.

SAbbas2018 committed:
Enabled CORS to allow calls from frontend to the backend api

SAbbas2018 committed:
Manually Ran Linters

jcserv reviewed (changes requested):
Tested it on my end, looks good just need to fix the linting changes!

SAbbas2018 committed:
Merge branch 'develop' into DEV-031

SAbbas2018 committed:
Enabled CORS to allow calls from frontend to the backend api

fix package-lock

Hotfix: Fix local builds failing

fix package-lock
Manually Ran Linters

Merge branch 'develop' into DEV-031

SAbbas2018 committed:
Merge branch 'DEV-031' of https://github.com/white-van/discussion-board into DEV-031

krishchow commented:
Can you squash these commits?

krishchow commented on README.md:
```diff
@@ -54,9 +54,11 @@ discussion-board follows Gitflow
+4. Fill out the pull request template accordingly
+5. To be approved, code must have adequate test coverage + formatted properly
+
 - Check frontend formatting with `npm run lint` and `make prettier`
```
Should this be 6?

krishchow commented on backend/README.md:
```diff
@@ -82,3 +82,8 @@ specificUserRouter := router.Group("/user/:uid"){
+
+### CORS
+
+Currently we have given access to http://localhost:3000 as an origin source for making calls to the backend. Wehn pushing the app
```
```suggestion
Currently we have given access to http://localhost:3000 as an origin source for making calls to the backend. When pushing the app
```

krishchow replied:
I think we can also add https://dialog-gamma.vercel.app/, right @jcserv?

krishchow commented on backend/httpd/main.go:
```diff
@@ -1,14 +1,33 @@
+	// the url for where ever we host our front end.
+	// All RESTful api methods are allowed
+	r.Use(cors.New(cors.Config{
+		AllowOrigins:     []string{"http://localhost:3000"},
```
```suggestion
		AllowOrigins:     []string{"http://localhost:3000", "https://dialog-gamma.vercel.app/"},
```
